src/Data/TypeRep/TMap.hs

```diff
@@ -1,89 +0,0 @@
-{-# LANGUAGE AllowAmbiguousTypes #-}
-{-# LANGUAGE Rank2Types          #-}
-
--- | This module introduces 'TMap' data type — 'TR.TypeRepMap' parametrized by 'Identity'.
-
-module Data.TypeRep.TMap
-       ( -- * Map type
-         TMap
-
-         -- * Construction
-       , empty
-       , one
-
-         -- * Modification
-       , insert
-       , delete
-       , hoist
-       , unionWith
-       , union
-
-         -- * Query
-       , lookup
-       , member
-       , size
-       ) where
-
-import Prelude hiding (lookup)
-
-import Data.Functor.Identity (Identity (..))
-import Data.Typeable (Typeable)
-import GHC.Exts (coerce)
-
-import qualified Data.TypeRep.Map as TR
-
--- | 'TR.TypeRepMap' parametrized by 'Identity'.
-type TMap = TR.TypeRepMap Identity
-
--- | Similar to 'TR.empty' but for 'TMap'.
-empty :: TMap
-empty = TR.empty
-{-# INLINE empty #-}
-
--- | Similar to 'TR.one' but for 'TMap'.
-one :: forall a . Typeable a => a -> TMap
-one x = coerce (TR.one @a @Identity $ coerce x)
-{-# INLINE one #-}
-
--- | Similar to 'TR.insert' but for 'TMap'.
-insert :: forall a . Typeable a => a -> TMap -> TMap
-insert x = coerce (TR.insert @a @Identity $ coerce x)
-{-# INLINE insert #-}
-
--- | Similar to 'TR.delete' but for 'TMap'.
-delete :: forall a . Typeable a => TMap -> TMap
-delete = TR.delete @a @Identity
-{-# INLINE delete #-}
-
--- | Similar to 'TR.hoist' but for 'TMap'.
-hoist :: (forall x. x -> g x) -> TMap -> TR.TypeRepMap g
-hoist f = TR.hoist (f . coerce)
-{-# INLINE hoist #-}
```

Comment:
Because this operation can't be fully specialized to `TMap`. In the current form it returns `TypeRepMap` (so there's no particular reason to duplicate it in `TMap`), and its fully specialized version would be

```
map :: (forall x. x -> x) -> TMap -> TMap
```

and it's completely useless.